src/middlewares/uniq.js

```diff
@@ -0,0 +1,23 @@
+import { SET, RESET } from '../actions'
+
+export const initState = { data: undefined, initialized: false }
+
+const reducer = (/* key */) => prefix =>
+  (state = initState, { type, payload } = {}) => {
+    switch (type) {
+      case SET(prefix):
+        return {
+          data: payload,
+          initialized: true,
```

Comment:
I think this is useful because you don't have to handle different kind of `data` type to know if this is set or no.

In an incoming PR, data could be initialized with a custom object (as @bpetetot asked for)